jwt/__main__.py

```diff
@@ -2,135 +2,172 @@
-                sys.exit(0)
-            except DecodeError as e:
-                print(e)
-                sys.exit(1)
-
-        # Try to encode
-        if options.key is None:
-            print('Key is required when encoding. See --help for usage.')
-            sys.exit(1)
-
-        # Build payload object to encode
-        payload = {}
-
-        for arg in arguments:
-            try:
-                k, v = arg.split('=', 1)
-
-                # exp +offset special case?
-                if k == 'exp' and v[0] == '+' and len(v) > 1:
-                    v = str(int(time.time()+int(v[1:])))
-
-                # Cast to integer?
-                if v.isdigit():
-                    v = int(v)
-                else:
-                    # Cast to float?
-                    try:
-                        v = float(v)
-                    except ValueError:
-                        pass
-
-                # Cast to true, false, or null?
-                constants = {'true': True, 'false': False, 'null': None}
-
-                if v in constants:
-                    v = constants[v]
-
-                payload[k] = v
-            except ValueError:
-                print('Invalid encoding input at {}'.format(arg))
-                sys.exit(1)
+    subparsers = arg_parser.add_subparsers(
+        title='PyJWT subcommands',
+        description='valid subcommands',
+        help='additional help'
+    )
 
-        try:
-            token = encode(
-                payload,
-                key=options.key,
-                algorithm=options.algorithm
-            )
+    # Encode subcommand
+    encode_parser = subparsers.add_parser('encode', help='use to encode a supplied payload')
+
+    payload_help = """Payload to encode. Must be a space separated list of key/value
+    pairs separated by equals (=) sign."""
+
+    encode_parser.add_argument('payload', nargs='+', help=payload_help)
+    encode_parser.set_defaults(func=encode_payload)
+
+    # Decode subcommand
+    decode_parser = subparsers.add_parser('decode', help='use to decode a supplied JSON web token')
+    decode_parser.add_argument('token', help='JSON web token to decode.')
+
+    decode_parser.add_argument(
+        '-n', '--no-verify',
+        action='store_false',
+        dest='verify',
+        default=True,
+        help='ignore signature and claims verification on decode'
+    )
+
+    decode_parser.set_defaults(func=decode_payload)
 
-            print(token)
-            sys.exit(0)
-        except Exception as e:
-            print(e)
-            sys.exit(1)
-    else:
-        p.print_help()
+    return arg_parser
+
+
+def main(args):
+    arg_parser = build_argparser()
+
+    try:
+        arguments = arg_parser.parse_args(args)
+
+        return arguments.func(arguments)
+    except Exception as e:
+        print('There was an unforseen error: ', e)
+        arg_parser.print_help()
+        sys.exit(1)
 
 
 if __name__ == '__main__':

```

Comment:
@froi Noticed another thing missing.

1. Run `python setup.py install`
2. Run `pyjwt`

It'll fail with something like this:

```
$ pyjwt
Traceback (most recent call last):
  File "/Users/jpadilla/.pyenv/versions/pyjwt/bin/pyjwt", line 9, in <module>
    load_entry_point('PyJWT==1.5.0', 'console_scripts', 'pyjwt')()
TypeError: main() takes exactly 1 argument (0 given)
```

Since we're using the `console_scripts` entry point on our `setup.py` I think we should be able to drop just drop the `if __name__ == '__main__': ...` and have the `main()` method optionally set args to `sys.argv[1:]`. Let me know if this makes sense.